internal/syncasync/sync_async_bridge.go

```diff
@@ -225,18 +242,23 @@ func (sa *syncAsyncBridge) getOperationFromEvent(event *fftypes.EventDelivery) (
 }
 
 func (sa *syncAsyncBridge) handleMessageConfirmedEvent(event *fftypes.EventDelivery) error {
+
+	// See if the CID marks this as a reply to an inflight message
+	inflight := sa.getInFlight(event.Namespace, messageConfirm, event.Reference)
+	inflightReply := sa.getInFlight(event.Namespace, messageReply, event.Correlator)
+
+	if inflightReply == nil && inflight == nil {
```

Comment:
I found as I was extending, that the sync async manager was needing to perform a DB lookup for any event emitted in the system - regardless of if anything was in-flight. This didn't feel right, and as I was working on the code here I thought I'd tweak it to check the in-flight list first.

This did mean adding a `correlator` field to the event, alongside the `reference`, that contains the `cid` from a reply message, or for definitions it's filled in by the definition handler appropriately.